sky/engine/core/dart/painting.dart

```diff
@@ -368,14 +393,16 @@ class Gradient extends Shader {
                   List<Color> colors,
                   [List<double> colorStops = null,
                   TileMode tileMode = TileMode.clamp]) {
-    _constructor();
     _validateColorStops(colors, colorStops);
-    _initRadial(center, radius, colors, colorStops, tileMode.index);
+    final Int32List colorsBuffer = _encodeColorList(colors);
+    final Float32List colorStopsBuffer = colorStops == null ? null : new Float32List.fromList(colorStops);
+    _constructor();
+    _initRadial(center.x, center.y, radius, colorsBuffer, colorStopsBuffer, tileMode.index);
   }
-  void _initRadial(Point center, double radius, List<Color> colors, List<double> colorStops, int tileMode) native "Gradient_initRadial";
+  void _initRadial(double centerX, double centerY, double radius, Int32List colors, Float32List colorStops, int tileMode) native "Gradient_initRadial";
 
   static void _validateColorStops(List<Color> colors, List<double> colorStops) {
-    if (colorStops != null && (colors == null || colors.length != colorStops.length))
+    if (colorStops != null && colors.length != colorStops.length)
```

Comment:
I'll write a patch that goes through all the APIs and adds the appropriate null checks.